### PR TITLE
18w50a exploration 2

### PR DESCRIPTION
**Pull request by Darkhax**
This PR is a few mappings I noticed while reverse engineering some of the vanilla code, not much of a common or logical thread between what was mapped. 

- Bell model and rederer class names
- Class names for where village structure/component data is being registered
- Mapped some of the missing enums and fields in the block states properties class.
- Mapped some methods in entity related to checking spawn light levels, and checking if they're in contact with different water sources.

### CODE REVIEW

File: mappings/net/minecraft/block/enums/Attatchment.mapping [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CLASS bqc net/minecraft/block/enums/Attatchment
```
Comment (Gegy) [inline mappings/net/minecraft/block/enums/Attatchment.mapping]: `Attachment`?

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 2]
```diff
@@ -12,11 +12,13 @@ CLASS bqd net/minecraft/state/property/Properties
 	FIELD K FACING_HOPPER Lbqh;
 	FIELD L FACING_HORIZONTAL Lbqh;
 	FIELD M WALL_MOUNT_LOCAITON Lbqk;
+	FIELD N ATTATCHMENT Lbqk;
```
Comment (Gegy) [inline mappings/net/minecraft/state/property/Properties.mapping]: `ATTACHMENT` here too

File: mappings/net/minecraft/block/enums/Attatchment.mapping (hunk 1 above)
Comment (Darkhax) [reply mappings/net/minecraft/block/enums/Attatchment.mapping]: One day I will learn how to spell lol

File: mappings/net/minecraft/block/enums/Half.mapping [hunk 3]
```diff
@@ -1,4 +1,4 @@
-CLASS bql
+CLASS bql net/minecraft/block/enums/Half
```
Comment (asiekierka) [inline mappings/net/minecraft/block/enums/Half.mapping]: Half or a more specific Half?

**Review by asiekierka — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/block/enums/Half.mapping (hunk 3 above)
Comment (Darkhax) [reply mappings/net/minecraft/block/enums/Half.mapping]: It's just a generic top/bottom half of a block. I was initially going to name this calls BlockHalf, however another class has that name. The other class is used for describing blocks that exist as two separate blocks, like doors and tall grass. The class I have named is used to describe the half of a specific block, like trap doors and stairs.

Comment (shadowfacts) [reply mappings/net/minecraft/block/enums/Half.mapping]: Then both should probably be renamed. Having both BlockHalf and Half is confusing if you don't remember which one is which.

Comment (Darkhax) [reply mappings/net/minecraft/block/enums/Half.mapping]: Okay, how do these names sound?

BlockHalf -> DoubleBlockHalf
Half -> BlockHalf

Comment (Prospector) [reply mappings/net/minecraft/block/enums/Half.mapping]: better than any alternatives I've heard. Do double plants use the former as well?

Comment (Darkhax) [reply mappings/net/minecraft/block/enums/Half.mapping]: @Prospector The "TallPlantBlock" class uses it, I think it's all the same class there.

**Review by asiekierka — APPROVED** [review]

**Review by shadowfacts — APPROVED** [review]

**Pull request merged**